speechbrain/core.py

```diff
@@ -882,9 +883,14 @@ def fit_batch(self, batch):
                     self.optimizer.step()
                 self.optimizer.zero_grad()
                 self.optimizer_step += 1
-
+        if should_step:
+            self.on_fit_batch_end()
```

Comment:
I'm wondering if we ought to always call `on_fit_batch_end` and perhaps pass `should_step` as an argument. Someone might want to do something there, even if `should_step` is `False`.

Reply:
I think passing the loss would likely make sense too.

Reply:
If one were going to compute e.g. WER in the `on_fit_batch_end`, they would need both the `predictions` and `batch` as well.

Reply:
pushed, `on_fit_batch_end()` is now passed `batch, outputs, loss, should_step`

let me know what you think

Reply:
true, will do :+1:

Reply:
done